### PR TITLE
Support building with the system mbedtls.

### PR DESCRIPTION
**Pull request by orbea**
## Description

This contains two commits.

The first commit extends the `check_header` function in `qb.libs.sh`, before it was limited to checking only 4 headers at once and now it can check any number of headers. Additionally there is no change in behavior for the existing uses of this function, but it will be needed for the following commit.

The second commit allows building RetroArch with the system mbedtls instead of the builtin version, this is highly desirable for some distros and also can make it easier to debug RetroArch and mbedtls issues.

By default RetroArch will prefer the builtin mbedtls and the average user does not need to do anything different. SSL/mbedtls support can still be completely disabled with `--disable-ssl` and if users wish to build with the system mbedtls they can use `--disable-builtinmbedtls`. If the builtin mbedtls is disabled and the system mbedtls is not found `configure` will print a warning and then disable SSL/mbedtls support.

I tested that SSL/mbedtls support still works with the builtin mbedtls and that it now works with `mbedtls-2.6.1` installed on my system.

## Related Issues

Fixes https://github.com/libretro/RetroArch/issues/7639.

### CODE REVIEW

**Comment by orbea** [comment]
Oops, let me fix the travis failure...

**Comment by twinaphex** [comment]
OK, let me try pinging @bparker06 here since he was the one who initially added mbedtls in the first place.

**Comment by orbea** [comment]
I think I know how to fix it, just silly mistakes and then a typo... Wait for travis on this one.

**Comment by orbea** [comment]
Should be fixed now.

File: griffin/griffin.c [hunk 1]
```diff
@@ -1466,6 +1466,7 @@ SSL
 ============================================================ */
 #if defined(HAVE_SSL)
 #if defined(HAVE_NETWORKING)
+#if defined(HAVE_BUILTINMBEDTLS)
```
Comment (bparker06) [inline griffin/griffin.c]: I think this effectively disables SSL for all griffin builds besides the ones you explicitly added `HAVE_BUILTINMBEDTLS` for in this PR (3DS and OSX?). Doing it this way means all the other project files now also need modification, like all the msvc's and other stuff.

Comment (orbea) [reply griffin/griffin.c]: Also not that these build systems are already doing this for `-DHAVE_BUILTINMINIUPNPC`.

Comment (bparker06) [reply griffin/griffin.c]: Ok I thought it was enabled for at least MSVC but I guess not. Basically I think it just comes down to whether or not we want to require two ifdefs to get SSL to work correctly in that case or not. I'll leave it up to @twinaphex.

Comment (orbea) [reply griffin/griffin.c]: I still think they would need it for this regardless of griffin?

https://github.com/libretro/RetroArch/pull/7745/files#diff-bb3428e6a736248b18d6df7696f75b19R28

**Comment by bparker06** [comment]
FYI #7451 has been fixed.

**Comment by orbea** [comment]
@twinaphex When you have a chance can you make a decision on this PR?

Some things to note.

* This will require a new define (`-DHAVE_BUILTINGMBEDTLS`) for using SSL support with the builtin mbedtls and this PR hooks it up for all existing cases of `HAVE_SSL`.
* It will not change the default behavior.
* Using system dependencies instead of bundled dependencies is preferred or required by some distros.

**Comment by twinaphex** [comment]
Fine then, we'll use HAVE_BUILTINMBEDTLS for platforms where we need to bake in mbedtls.

**Pull request merged**